app/controllers/api/v1/packages_controller.rb

```diff
@@ -127,16 +127,11 @@ def print_barcode
       def search_stockit_items
         records = @packages # security
         records = records.undispatched if params["orderId"].present?
-        if params['searchText'].present?
-          records = records.search(
-            params['searchText'],
-            params['itemId'].presence,
-            with_inventory_no: params['withInventoryNumber'] == 'true'
-          )
-        end
-        # apply selected filters
-        records = apply_filters(records).page(params["page"]).per(params["per_page"])
-        pages = records.total_pages
+        records = records.search(search_text: params['searchText'], item_id: params['itemId'],
+          with_inventory_no: params['withInventoryNumber'] == 'true') if params['searchText'].present?
+        params_for_filter = ['state', 'location'].inject({}){|h, k| h[k] = params[k] if params[k].present?; h}
```

Comment:
Can we use each_with_object here so that we don't need to return `h` at the end of inject.

https://blog.arkency.com/inject-vs-each-with-object/

Reply:
Thanks, I'm really pleased to learn about this method... much nicer.